core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -1051,15 +1049,19 @@ private[spark] class TaskSetManager(
       logDebug("Task length threshold for speculation: " + threshold)
       for (tid <- runningTasksSet) {
         var speculated = checkAndSubmitSpeculatableTask(tid, time, threshold)
-        if (!speculated && tidToExecutorKillTimeMapping.contains(tid)) {
-          // Check whether this task will finish before the exectorKillTime assuming
-          // it will take medianDuration overall. If this task cannot finish within
-          // executorKillInterval, then this task is a candidate for speculation
-          val taskEndTimeBasedOnMedianDuration = taskInfos(tid).launchTime + medianDuration
-          val canExceedDeadline = tidToExecutorKillTimeMapping(tid) <
-            taskEndTimeBasedOnMedianDuration
-          if (canExceedDeadline) {
-            speculated = checkAndSubmitSpeculatableTask(tid, time, 0)
+        if (!speculated && executorDecommissionKillInterval.nonEmpty) {
```

Comment:
This code is only invoked with `speculationEnabled`, so the previous `speculationEnabled` enabled check added by @Ngone51 on line 1123 is not needed anymore.

Reply:
I see, we don't have `tidToExecutorKillTimeMapping` anymore.